resolwe_bio/processes/reads_processing/trimmomatic.yml

```diff
@@ -379,65 +468,139 @@
-        fw_unpaired_filtered.fastq.gz \
-        rw_paired_filtered.fastq.gz \
-        rw_unpaired_filtered.fastq.gz \
-        {% if adapters %}"ILLUMINACLIP:${ADAPTERS}.fasta:"{{ illuminaclip.seed_mismatches }}":"{{illuminaclip.palindrome_clip_threshold}}":"{{ illuminaclip.simple_clip_threshold }}":${ILLUMINACLIP_OPTIONAL}"{% endif %} \
-        {% if slidingwindow.window_size and slidingwindow.required_quality %}"SLIDINGWINDOW:"{{ slidingwindow.window_size }}":"{{ slidingwindow.required_quality }}{% endif %} \
-        {% if maxinfo.target_length and maxinfo.strictness %}}"MAXINFO:"{{maxinfo.target_length}}":"{{maxinfo.strictness}}{% endif %} \
-        {% if leading %}"LEADING:"{{ leading }}{% endif %} \
-        {% if trailing %}"TRAILING:"{{ trailing }}{% endif %} \
-        {% if crop %}"CROP:"{{ crop }}{% endif %} \
-        {% if headcrop %}"HEADCROP:"{{ headcrop }}{% endif %} \
-        {% if minlen %}"MINLEN:"{{minlen}}{% endif %} \
-        {% if average_quality %}"AVGQUAL:"{{ average_quality }}{% endif %}
-
-      re-checkrc "Failed while processing with Trimmomatic."
-      re-progress 0.7
-
-      mv fw_paired_filtered.fastq.gz "${NAME1}_paired_filtered.fastq.gz"
-      mv fw_unpaired_filtered.fastq.gz "${NAME1}_unpaired_filtered.fastq.gz"
-      mv rw_paired_filtered.fastq.gz "${NAME2}_paired_filtered.fastq.gz"
-      mv rw_unpaired_filtered.fastq.gz "${NAME2}_unpaired_filtered.fastq.gz"
-
-      echo "Postprocessing FastQC..."
-      # http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
-      mkdir "fastqc" && fastqc "${NAME1}_paired_filtered.fastq.gz" "${NAME2}_paired_filtered.fastq.gz" --extract --outdir="fastqc" >&2
-      re-checkrc "Failed while processing with FastQC."
-      re-progress 0.9
-
-      mv "fastqc/${NAME1}_paired_filtered_fastqc" "fastqc/${NAME1}_fastqc"
-      mv "fastqc/${NAME2}_paired_filtered_fastqc" "fastqc/${NAME2}_fastqc"
-      mv "fastqc/${NAME1}_paired_filtered_fastqc.zip" "${NAME1}_fastqc.zip"
-      mv "fastqc/${NAME2}_paired_filtered_fastqc.zip" "${NAME2}_fastqc.zip"
-
-      FASTQC_URL="{\"file\":\"fastqc/${NAME1}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME1}_fastqc\"]}"
-      FASTQC_URL2="{\"file\":\"fastqc/${NAME2}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME2}_fastqc\"]}"
-
-      re-save-file-list fastq "${NAME1}_paired_filtered.fastq.gz"
-      re-save-file-list fastq2 "${NAME2}_paired_filtered.fastq.gz"
-      re-save-file stats stats.txt
-
-      if [ -f "${NAME1}_unpaired_filtered.fastq.gz" ]; then
-        re-save-file fastq_unpaired "${NAME1}_unpaired_filtered.fastq.gz"
-      fi
-
-      if [ "-f ${NAME2}_unpaired_filtered.fastq.gz" ]; then
-        re-save-file fastq2_unpaired "${NAME2}_unpaired_filtered.fastq.gz"
-      fi
-
-      re-save-file-list fastqc_archive "${NAME1}_fastqc.zip"
-      re-save-file-list fastqc_archive2 "${NAME2}_fastqc.zip"
-      re-save-list fastqc_url "${FASTQC_URL}"
-      re-save-list fastqc_url2 "${FASTQC_URL2}"
+      {% if
```

Comment:
The condition check is incorrect. If the user specifies just e.g. `adapters` this warning will not be displayed but the reads will still be copied from the input to the output.

Reply:
reads field is reqired